### PR TITLE
Remove `include_type_name` parameter from REST layer

### PR DESCRIPTION
**Pull request by romseygeek**
This commit removes support for the `include_type_name` parameter from all
REST actions that receive or return mapping configurations.

Relates to #41059

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/admin/indices/rollover/RolloverRequest.java [hunk 1]
```diff
@@ -70,20 +68,14 @@
...
-                    throw new IllegalArgumentException("The mapping definition cannot be nested under a type " +
-                        "[" + MapperService.SINGLE_MAPPING_NAME + "] unless include_type_name is set to true.");
-                }
-                request.createIndexRequest.mapping(MapperService.SINGLE_MAPPING_NAME, mappings);
+        PARSER.declareField((parser, request, context) -> {
+            // a type is not included, add a dummy _doc type
+            Map<String, Object> mappings = parser.map();
+            if (MapperService.isMappingSourceTyped(MapperService.SINGLE_MAPPING_NAME, mappings)) {
+                throw new IllegalArgumentException("The mapping definition cannot be nested under a type " +
+                    "[" + MapperService.SINGLE_MAPPING_NAME + "] unless include_type_name is set to true.");
```
Comment (mayya-sharipova) [inline server/src/main/java/org/elasticsearch/action/admin/indices/rollover/RolloverRequest.java]: As we don't have `include_type_name` parameter anymore, I think the error message should just say: "The mapping definition cannot be nested under a type."

File: server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestCreateIndexAction.java [hunk 2]
```diff
@@ -77,9 +65,8 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
     }
 
 
-    static Map<String, Object> prepareMappings(Map<String, Object> source, boolean includeTypeName) {
-        if (includeTypeName
-            || source.containsKey("mappings") == false
+    static Map<String, Object> prepareMappings(Map<String, Object> source) {
+        if (source.containsKey("mappings") == false
```
Comment (mayya-sharipova) [inline server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestCreateIndexAction.java]: here also on the line 80, may be also worth correcting  the error message to have only "The mapping definition cannot be nested under a type"

**Review by mayya-sharipova — APPROVED** [review]
@romseygeek, this PR LGTM. 
I have just a small comment about the error message, which can also be addressed in next PRs  of type  removals.

**Comment by romseygeek** [comment]
@elasticmachine update branch

**Pull request merged**